patchset: add xcam video filter

liuyinhangx opened the pull request:
Apply patch:
$ git am patchset/*.patch

libXCam supports automotive surround view stitching, 360 video stitching,
digital video stabilization, noise reduction and so on.
For more information about libXCam see https://github.com/intel/libxcam.

To enable compilation of xcam video filter you need to configure FFmpeg
with --enable-libxcam.

For more detailed test cases see
https://github.com/intel/libxcam/wiki/Tests#1-ffmpeg-xcam.

Signed-off-by: Yinhang Liu <yinhangx.liu@intel.com>

guoyejun commented on patchset/0057-avfilter-vf_xcam-add-xcam-video-filter.patch:
```diff
@@ -0,0 +1,508 @@
++++ b/libavfilter/vf_xcam.c
+@@ -0,0 +1,317 @@
++/*
++ * Copyright (c) 2020 Intel Corporation, all rights reserved.
```
shall  we add this line?

guoyejun commented on patchset/0057-avfilter-vf_xcam-add-xcam-video-filter.patch:
```diff
@@ -0,0 +1,508 @@
++{
++    if (avfmt == AV_PIX_FMT_YUV420P)
++        return V4L2_PIX_FMT_YUV420;
++    return V4L2_PIX_FMT_NV12;
```
what happens if avfmt is not nv12

liuyinhangx replied:
xcam video filter only supports yuv420 and nv12 formats, and the formats have been limited to [yuv420, nv12] in xcam_query_formats().

liuyinhangx committed:
patchset: add xcam video filter

Apply patches:
$ git am patchset/0057-avfilter-vf_xcam-add-xcam-video-filter.patch

libXCam supports automotive surround view stitching, 360 video stitching,
digital video stabilization, noise reduction and so on.
For more information about libxcam see https://github.com/intel/libxcam.

Please refer to https://github.com/intel/libxcam/wiki/Build to build
libxcam.

To enable compilation of xcam video filter, please configure FFmpeg with
--enable-libxcam.

Here are the features provided by xcam video filter:
- stitch     CPU|GLES|Vulkan stitching
- stitchcl   OpenCL stitching
- fisheye    Fisheye calibration
- 3dnr       3D denoising
- waveletnr  Wavelet denoising
- dvs        Digital video stabilization
- defog      Fog removal
Among these features, CPU stitching supports AVX512 acceleration and
both NV12 and YUV420 pixel formats, all the other features support GPU
acceleration and NV12 pixel format.
stitch and stitchcl features support dynamic inputs, fisheye, 3dnr,
waveletnr, dvs and defog features support one input.

Use 'ffmpeg -h filter=xcam' to get the common parameters, stitch and
stitchcl have private parameters, use 'params=help=1' to get the private
parameters. For more detailed test cases see
https://github.com/intel/libxcam/wiki/Tests#1-ffmpeg-xcam.

Signed-off-by: Yinhang Liu <yinhangx.liu@intel.com>

uartie commented:
Patch breaks cartwheel build... patch does not apply.  Please fix and reopen if needed.
